SampleApps/QuickStart-SwiftUI/QuickStart/ViewModels/NodeViewModel.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  NodeViewModel.swift
+//  QuickStart
+//
+// Copyright (c) 2023 ForgeRock. All rights reserved.
+//
+// This software may be modified and distributed under the terms
+// of the MIT license. See the LICENSE file for details.
+//
+
+import Foundation
+import FRAuth
+
+@MainActor class NodeViewModel: ObservableObject {
+    @Published var callbackViewModels = [CallbackViewModel]()
+    @Published var showNodeView = false
+    @Published var shouldDismiss: Bool = false
+    
+    var statusViewModel: StatusViewModel?
+    var journeyName: String = "\(FRAuth.shared?.options?.authServiceName ?? "") Journey" 
+    var node: Node?
+    
+    init(node: Node? = nil) {
+        self.node = node
+    }
+    
+    func parseNode() {
+        guard let node = node else {
+            print("Node is missing")
+            return
+        }
+        
+        for callback in node.callbacks {
+            if callback.isKind(of: NameCallback.self), let nameCallBack = callback as? NameCallback {
+                callbackViewModels.append(CallbackViewModel(name: nameCallBack.prompt!, value: ""))
+            } else if callback.isKind(of: PasswordCallback.self), let passwordCallback = callback as? PasswordCallback {
+                callbackViewModels.append(CallbackViewModel(name: passwordCallback.prompt!, value: "", isSecret: true))
+            } else {
+                statusViewModel?.status = Status(statusDescription: "\(callback.type) is not suppoorted", statusType: .error)
+                callbackViewModels.removeAll()
+            }
+        }
+        
+        if callbackViewModels.count > 0 {
+            statusViewModel?.status = Status(statusDescription: "Node successfully parsed", statusType: .info)
+        }
+        
+    }
+    
+    func submitNode() {
+        guard let node = node, let callbacks = node.callbacks as? [SingleValueCallback] else { return }
```

Comment:
I am not 100% sure about this. It works on this example, but the moment you add more callback types it will need refactoring

Reply:
We check earlier i the code for supported Callbacks and show error if we encounter a callback that is not supported. That being said I moved the callback type casting/checking down in the for loop. safer this way